theano/tensor/raw_random.py

```diff
@@ -133,15 +133,28 @@ def __init__(self, fn, outtype, inplace=False, ndim_added=0):
         self.__setstate__([fn, outtype, inplace, ndim_added])
 
     def __getstate__(self):
-        return self.state
+        d = dict(self.__dict__)
+        del d['exec_fn']
+        if 'destroy_map' in d:
+            del d['destroy_map']
+        return d
+
+    def __setstate__(self, dct):
+        if isinstance(dct, dict):
+            state = [dct['fn'],
+                     dct['outtype'],
+                     dct['inplace'],
+                     dct['ndim_added']]
+        else:
+            state = dct
```

Comment:
done.

On Wed, Mar 30, 2016 at 3:48 PM, Pascal Lamblin <notifications@github.com>
wrote:

> In theano/tensor/raw_random.py
> <https://github.com/Theano/Theano/pull/4310#discussion_r57951541>:
>
> > @@ -133,15 +133,28 @@ def __init__(self, fn, outtype, inplace=False, ndim_added=0):
> >          self.__setstate__([fn, outtype, inplace, ndim_added])
> >
> >      def __getstate__(self):
> > -        return self.state
> > +        d = dict(self.__dict__)
> > +        del d['exec_fn']
> > +        if 'destroy_map' in d:
> > +            del d['destroy_map']
> > +        return d
> > +
> > +    def __setstate__(self, dct):
> > +        if isinstance(dct, dict):
> > +            state = [dct['fn'],
> > +                     dct['outtype'],
> > +                     dct['inplace'],
> > +                     dct['ndim_added']]
> > +        else:
> > +            state = dct
>
> Oh, right, we should call self.__dict__.update(...).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4310/files/e063f64bc59cbbf32ffe5ca13b09ddc7f142d3e8#r57951541>
>